### ISSUE TITLE
If load_user_from_database' is set to false, there is no way to get user info from db ?

### ISSUE THREAD

**Issue by raj-saroj-vst-au4** [robsontenorio/laravel-keycloak-guard#108]
If load_user_from_database is set to false, and i try auth()->user() it throws an empty array.

Is there any specific method available that i can call to get the user data from db ?
my flow is as follows :- 
the user authenticates through keycloak > is forced on to signup page(due to a middleware that checks for user auth and signupcomplete column) > user signs up > records in db and sets signupcomplete to 1 

now since i have set the load_user_from_database to false how do i get user info from db like auth()->user() or any other way ?

**Comment by robsontenorio** [comment]
If you set it to “false” of course this package won’t try to load the user from the database. 

So , you need to do it by yourself using some middleware

**Comment by raj-saroj-vst-au4** [comment]
but if the user is logging in for the first time, how would db contain user data to be able to pull his info ?
and we cant even have an unprotected api route cause of security issues.

**Comment by raj-saroj-vst-au4** [comment]
can u make this method more modular so that i can call it explicitly ?
KeycloakGuard::class->validate()

for example

$loadfromdb = $this->config['load_user_from_database'];

public function validate(array $credentials = [], $loadfromdb = false)
    {
        $this->validateResources();

        if ($loadfromdb) {
            $methodOnProvider = $this->config['user_provider_custom_retrieve_method'] ?? null;

            if ($methodOnProvider) {
                $user = $this->provider->{$methodOnProvider}($this->decodedToken, $credentials);
            } else {
                $user = $this->provider->retrieveByCredentials($credentials);
            }

            if (!$user) {
                throw new UserNotFoundException("User not found. Credentials: ".json_encode($credentials));
            }
        } else {
            $class = $this->provider->getModel();
            $user = new $class();
        }

        $this->setUser($user);

        return true;
    }

**Comment by robsontenorio** [comment]
Take a look at “user_provider_custom_retrieve_method” on README.

**Comment by raj-saroj-vst-au4** [comment]
Yeah i read that but, that part would execute only if the load_user_from_db is true…
